### ISSUE TITLE
Support Sass namespaced variables

### ISSUE THREAD

**Issue by thibaudcolas** [AndyOGo/stylelint-declaration-strict-value#52]
Sass now has a [module system with `@use` and `@forward`](https://sass-lang.com/documentation/at-rules/use). With this new syntax, we can use variables and functions like:

```scss
@use '@primer/css/support/index' as primer;

.btn {
    color: primer.$text-gray-dark;
    background: primer.light-gradient()
```

Currently I get the following error: `Expected variable, function or keyword for "primer.$text-gray-dark" of "color"`.

---

Looking at the code, this is because the variables regex doesn’t account for that new syntax:

https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/197e257fb528015496b9d9ea4e062b7a28896053/src/index.js#L13-L15

Interestingly the functions regex already supports the new syntax 😄

Thinking of how to solve this – this is the regex I came up with, trying my best to add the new pattern without making it all slower by introducing overlaps between the patterns:

```regex
/^-?(?:@.+|\$.+|var\(--.+\)|(?!@|\$|var\(--).+\.\$.+)$/
```

https://regexper.com/#%2F%5E-%3F%28%3F%3A%40.%2B%7C%5C%24.%2B%7Cvar%5C%28--.%2B%5C%29%7C%28%3F!%40%7C%5C%24%7Cvar%5C%28--%29.%2B%5C.%5C%24.%2B%29%24%2F

![regexper](https://user-images.githubusercontent.com/877585/79689497-36b2c680-824d-11ea-8594-40a1760a5daa.png)

If that sounds like a viable approach I would be happy to make a PR for this, testing the performance a bit further and writing tests for the new pattern.

**Comment by AndyOGo** [comment]
@thibaudcolas Thanks for your issue 👍 

This is super easy to add, I just made a PR myself.

**Comment by thibaudcolas** [comment]
Wow, that was fast @AndyOGo! I’ll follow up on the PR.

**Comment by AndyOGo** [comment]
Thank you a lot @thibaudcolas 
I just published [`v1.2.0`](https://www.npmjs.com/package/stylelint-declaration-strict-value/v/1.2.0).

**Comment by thibaudcolas** [comment]
Thank you! I’ve just tried it on my project and it’s working like a charm.
